state/state.go

```diff
@@ -27,3 +33,38 @@ type View interface {
 	NewView(ctx context.Context, changes merkledb.ViewChanges) (merkledb.View, error)
 	GetMerkleRoot(ctx context.Context) (ids.ID, error)
 }
+
+// StateKey holds the name of the key and its permission (Read/Write)
+// Specifying RWrite is setting both the Read and Write bit
+type Key struct {
+	Name string
+	// TODO: consider a dynamically sized []byte
+	Permission Permission
+}
+
+// TODO: Support Alloc bit
+type Permission [PermissionLen]byte
+
+// Returns a StateKey with the appropriate permission bits set
+// By default, no permission bits are set
+func NewKey(name string, permissions ...int) Key {
+	var key Key
+	key.Name = name
+	for _, perm := range permissions {
+		// Only set bit if it's within the byte slice
```

Comment:
You are setting bytes here, not bits.

Reply:
I agree, we should set bits.

Reply:
We have some utilities for this: https://github.com/ava-labs/avalanchego/blob/master/utils/set/bits.go

Reply:
Related: https://github.com/ava-labs/hypersdk/issues/705